subsys/net/ip/net_if.c

```diff
@@ -1418,10 +1424,18 @@ const struct in6_addr *net_if_ipv6_select_src_addr(struct net_if *dst_iface,
 
 u32_t net_if_ipv6_calc_reachable_time(struct net_if *iface)
 {
-	return MIN_RANDOM_FACTOR * iface->ipv6.base_reachable_time +
-		sys_rand32_get() %
-		(MAX_RANDOM_FACTOR * iface->ipv6.base_reachable_time -
-		 MIN_RANDOM_FACTOR * iface->ipv6.base_reachable_time);
+	u32_t min_reachable, max_reachable;
+
+	min_reachable = MIN_RANDOM_NUMER *
```

Comment:
Hrm, it might be better to do the multiplication first and then divide.  I'd be open to re-posting this with the math reversed.